### PR TITLE
Error for global symbolic variables are assigned 0 - Issue #28 

### PR DESCRIPTION
**Pull request by Himeshi**
Depends on pull request #31.

### CODE REVIEW

**Comment by domainexpert** [comment]
@Himeshi Is this PR a work in progress not yet ready for merging? In such case I would usually prefix the title with "WIP:" to clarify. Please feel free to assign me as a reviewer when this is ready for review.

**Comment by Himeshi** [comment]
@domainexpert I think it is ready - any changes you wish to see in this? 

The function `isInStoredError` is implemented in pull request #31 but a quick Google search told me that there is no way to create dependent pull requests.

**Comment by domainexpert** [comment]
@Himeshi I recommend we continue working on this after #31 is merged.

File: lib/Core/Executor.cpp [hunk 1]
```diff
@@ -1112,7 +1112,10 @@ const Cell& Executor::eval(KInstruction *ki, unsigned index,
   if (vnumber < 0) {
     unsigned index = -vnumber - 2;
     Cell &ret = kmodule->constantTable[index];
-    ret.error = ConstantExpr::create(0, Expr::Int8);
+    if(state.symbolicError->checkStoredError(ret.value))
+         ret.error = state.symbolicError->retrieveStoredError(ret.value);
+    else
+         ret.error = ConstantExpr::create(0, Expr::Int8);
```
Comment (domainexpert) [inline lib/Core/Executor.cpp]: @Himeshi This PR has some indentation issues. Have you applied clang-format?

File: lib/Core/SymbolicError.h [hunk 2]
```diff
@@ -105,6 +105,14 @@ class SymbolicError {
     return errorState->retrieveError(value);
   }
 
+  bool checkStoredError(ref<Expr> address) {
+	  return errorState->isInStoredError(address);
```
Comment (domainexpert) [inline lib/Core/SymbolicError.h]: @Himeshi Indentation issue here (too many indent spaces).

File: lib/Core/SymbolicError.h [hunk 3]
```diff
@@ -105,6 +105,14 @@ class SymbolicError {
...
+  }
+
+  ref<Expr> retrieveStoredError(ref<Expr> address) {
+	  return errorState->retrieveStoredError(address);
```
Comment (domainexpert) [inline lib/Core/SymbolicError.h]: @Himeshi Ditto here.

**Review by domainexpert — CHANGES_REQUESTED** [review]
@Himeshi Thank you for the good work. After adding new commits, you could rebase with the `master`, as follows:
```
git checkout master
git pull
git checkout issue28
git rebase -i master
```

**Review by domainexpert — APPROVED** [review]

**Pull request merged**